### PR TITLE
PROD-2815 strip script and style tags for the terms and privacy policy popup

### PR DESCRIPTION
**Pull request by bb-yudhisthir**
strip script and style tags for the terms and privacy policy popup

[PROD-2815]
PROD-2815 #comment strip script and style tags for the terms and privacy policy popup

### Jira Issue:

https://buddyboss.atlassian.net/browse/PROD-2815

### General Note
Keep all conversations related to this PR in the associated Jira issue(s). Do NOT add comment on this PR or edit this PR’s description.

### Notes to Developer
* Ensure the IDs (i.e. PROD-1) of all associated Jira issues are reference in this PR’s title
* Ensure that you have achieved the Definition of Done before submitting for review
* When this PR is ready for review, move the associate Jira issue(s) to “Needs Review” (or “Code Review” for Dev Tasks)

### Notes to Reviewer
* Ensure that the Definition of Done have been achieved before approving a PR
* When this PR is approved, move the associated Jira issue(s) to “Needs QA” (or “Approved” for Dev Tasks)

[PROD-2815]: https://buddyboss.atlassian.net/browse/PROD-2815?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/bp-core/bp-core-catchuri.php [hunk 1]
```diff
@@ -1522,7 +1522,8 @@ function bp_core_change_privacy_policy_link_on_private_network( $link, $privacy_
 
 		if ( $privacy_policy_url && $page_title ) {
 			$get_privacy_policy = get_post( $policy_page_id );
-			$get_content        = apply_filters( 'bp_privacy_policy_content', wp_kses_post( apply_filters( 'the_content', $get_privacy_policy->post_content ) ), $get_privacy_policy->post_content );
+			$get_content        = bp_strip_script_and_style_tags( apply_filters( 'the_content', $get_privacy_policy->post_content ) );
+			$get_content        = apply_filters( 'bp_privacy_policy_content', wp_kses_post( $get_content ), $get_privacy_policy->post_content );
```
Comment (jitendrabanjara1991) [inline src/bp-core/bp-core-catchuri.php:1526]:
```suggestion
			$get_content        = apply_filters( 'bp_privacy_policy_content', wp_kses_post( bp_strip_script_and_style_tags( apply_filters( 'the_content', $get_privacy_policy->post_content ) ) ), $get_privacy_policy->post_content );
```
And remove 1525 line

File: src/bp-templates/bp-nouveau/includes/template-tags.php [hunk 2]
```diff
@@ -2608,15 +2608,17 @@ function bp_nouveau_signup_terms_privacy() {
 			<h1><?php echo esc_html( get_the_title( $terms ) ); ?></h1>
 			<?php
 			$get_terms = get_post( $terms );
-			echo apply_filters( 'the_content', $get_terms->post_content );
+			$get_term_content = bp_strip_script_and_style_tags( apply_filters( 'the_content', $get_terms->post_content ) );
+			echo apply_filters( 'bp_term_of_service_content', wp_kses_post( $get_term_content ), $get_terms->post_content );
```
Comment (jitendrabanjara1991) [inline src/bp-templates/bp-nouveau/includes/template-tags.php:2612]:
```suggestion
			echo apply_filters( 'bp_term_of_service_content', wp_kses_post( bp_strip_script_and_style_tags( apply_filters( 'the_content', $get_terms->post_content ) ) ), $get_terms->post_content );
```
And remove 2611 line

**Review by jitendrabanjara1991 — CHANGES_REQUESTED** [review]

File: src/bp-core/bp-core-catchuri.php [hunk 3]
```diff
@@ -1522,7 +1522,7 @@ function bp_core_change_privacy_policy_link_on_private_network( $link, $privacy_
 
 		if ( $privacy_policy_url && $page_title ) {
 			$get_privacy_policy = get_post( $policy_page_id );
-			$get_content        = apply_filters( 'bp_privacy_policy_content', wp_kses_post( apply_filters( 'the_content', $get_privacy_policy->post_content ) ), $get_privacy_policy->post_content );
+			$get_content        = apply_filters( 'bp_privacy_policy_content', wp_kses_post( bp_strip_script_and_style_tags( apply_filters( 'the_content', $get_privacy_policy->post_content ) ) ), $get_privacy_policy->post_content );
```
Comment (chetansatasiya) [inline src/bp-core/bp-core-catchuri.php:1525]:
```suggestion
			$get_content        = apply_filters( 'bp_privacy_policy_content', apply_filters( 'the_content', $get_privacy_policy->post_content ), $get_privacy_policy->post_content );
```

File: src/bp-core/bp-core-catchuri.php [hunk 4]
```diff
@@ -1538,7 +1538,7 @@ function bp_core_change_privacy_policy_link_on_private_network( $link, $privacy_
 		if ( ! empty( $terms ) && ! empty( $privacy ) ) {
 			$page_title       = ! empty( $terms ) ? get_the_title( $terms ) : '';
 			$get_terms        = get_post( $terms );
-			$get_term_content = apply_filters( 'bp_term_of_service_content', wp_kses_post( apply_filters( 'the_content', $get_terms->post_content ) ), $get_terms->post_content );
+			$get_term_content = apply_filters( 'bp_term_of_service_content', wp_kses_post( bp_strip_script_and_style_tags( apply_filters( 'the_content', $get_terms->post_content ) ) ), $get_terms->post_content );
```
Comment (chetansatasiya) [inline src/bp-core/bp-core-catchuri.php:1541]:
```suggestion
			$get_term_content = apply_filters( 'bp_term_of_service_content', apply_filters( 'the_content', $get_terms->post_content ), $get_terms->post_content );
```

File: src/bp-templates/bp-nouveau/includes/template-tags.php [hunk 5]
```diff
@@ -2640,7 +2640,7 @@ function bp_nouveau_signup_terms_privacy() {
 			<h1><?php echo esc_html( get_the_title( $privacy ) ); ?></h1>
 			<?php
 			$get_privacy = get_post( $privacy );
-			echo apply_filters( 'the_content', $get_privacy->post_content );
+			echo apply_filters( 'bp_privacy_policy_content', wp_kses_post( bp_strip_script_and_style_tags( apply_filters( 'the_content', $get_privacy->post_content ) ) ), $get_privacy->post_content );
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/template-tags.php:2643]:
```suggestion
			echo apply_filters( 'bp_privacy_policy_content', apply_filters( 'the_content', $get_privacy->post_content ), $get_privacy->post_content );
```

File: src/bp-templates/bp-nouveau/includes/template-tags.php [hunk 6]
```diff
@@ -2608,15 +2608,15 @@ function bp_nouveau_signup_terms_privacy() {
...
+			echo apply_filters( 'bp_term_of_service_content', wp_kses_post( bp_strip_script_and_style_tags( apply_filters( 'the_content', $get_terms->post_content ) ) ), $get_terms->post_content );
 			?>
 			<button title="<?php esc_attr_e( 'Close (Esc)', 'buddyboss' ); ?>" type="button" class="mfp-close"><?php esc_html_e( '×', 'buddyboss' ); ?></button>
 		</div>
 		<div id="privacy-modal" class="mfp-hide registration-popup bb-modal">
 			<h1><?php echo esc_html( get_the_title( $privacy ) ); ?></h1>
 			<?php
 			$get_privacy = get_post( $privacy );
-			echo apply_filters( 'the_content', $get_privacy->post_content );
+			echo apply_filters( 'bp_privacy_policy_content', wp_kses_post( bp_strip_script_and_style_tags( apply_filters( 'the_content', $get_privacy->post_content ) ) ), $get_privacy->post_content );
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/template-tags.php:2619]:
```suggestion
			echo apply_filters( 'bp_privacy_policy_content', apply_filters( 'the_content', $get_privacy->post_content ), $get_privacy->post_content );
```

File: src/bp-templates/bp-nouveau/includes/template-tags.php [hunk 7]
```diff
@@ -2608,15 +2608,15 @@ function bp_nouveau_signup_terms_privacy() {
 			<h1><?php echo esc_html( get_the_title( $terms ) ); ?></h1>
 			<?php
 			$get_terms = get_post( $terms );
-			echo apply_filters( 'the_content', $get_terms->post_content );
+			echo apply_filters( 'bp_term_of_service_content', wp_kses_post( bp_strip_script_and_style_tags( apply_filters( 'the_content', $get_terms->post_content ) ) ), $get_terms->post_content );
```
Comment (chetansatasiya) [inline src/bp-templates/bp-nouveau/includes/template-tags.php:2611]:
```suggestion
			echo apply_filters( 'bp_term_of_service_content', apply_filters( 'the_content', $get_terms->post_content ), $get_terms->post_content );
```

File: src/bp-core/bp-core-catchuri.php [hunk 8]
```diff
@@ -1577,7 +1577,7 @@ function bp_core_change_privacy_policy_link_on_private_network( $link, $privacy_
 		} elseif ( ! empty( $terms ) && empty( $privacy ) ) {
 			$page_title       = ! empty( $terms ) ? get_the_title( $terms ) : '';
 			$get_terms        = get_post( $terms );
-			$get_term_content = apply_filters( 'bp_term_of_service_content', wp_kses_post( apply_filters( 'the_content', $get_terms->post_content ) ), $get_terms->post_content );
+			$get_term_content = apply_filters( 'bp_term_of_service_content', wp_kses_post( bp_strip_script_and_style_tags( apply_filters( 'the_content', $get_terms->post_content ) ) ), $get_terms->post_content );
```
Comment (chetansatasiya) [inline src/bp-core/bp-core-catchuri.php:1580]:
```suggestion
			$get_term_content = apply_filters( 'bp_term_of_service_content', apply_filters( 'the_content', $get_terms->post_content ), $get_terms->post_content );
```

File: src/bp-core/bp-core-catchuri.php [hunk 9]
```diff
@@ -1564,7 +1564,7 @@ function bp_core_change_privacy_policy_link_on_private_network( $link, $privacy_
 		} elseif ( empty( $terms ) && ! empty( $privacy ) ) {
 			$page_title  = ! empty( $privacy ) ? get_the_title( $privacy ) : '';
 			$get_privacy = get_post( $privacy );
-			$get_content = apply_filters( 'bp_privacy_policy_content', wp_kses_post( apply_filters( 'the_content', $get_privacy->post_content ) ), $get_privacy->post_content );
+			$get_content = apply_filters( 'bp_privacy_policy_content', wp_kses_post( bp_strip_script_and_style_tags( apply_filters( 'the_content', $get_privacy->post_content ) ) ), $get_privacy->post_content );
```
Comment (chetansatasiya) [inline src/bp-core/bp-core-catchuri.php:1567]:
```suggestion
			$get_content = apply_filters( 'bp_privacy_policy_content', apply_filters( 'the_content', $get_privacy->post_content ), $get_privacy->post_content );
```

File: src/bp-core/bp-core-catchuri.php [hunk 10]
```diff
@@ -1550,7 +1550,7 @@ function bp_core_change_privacy_policy_link_on_private_network( $link, $privacy_
 			);
 			$page_title       = ( $privacy ) ? get_the_title( $privacy ) : '';
 			$get_privacy      = get_post( $privacy );
-			$get_content      = apply_filters( 'bp_privacy_policy_content', wp_kses_post( apply_filters( 'the_content', $get_privacy->post_content ) ), $get_privacy->post_content );
+			$get_content      = apply_filters( 'bp_privacy_policy_content', wp_kses_post( bp_strip_script_and_style_tags( apply_filters( 'the_content', $get_privacy->post_content ) ) ), $get_privacy->post_content );
```
Comment (chetansatasiya) [inline src/bp-core/bp-core-catchuri.php:1553]:
```suggestion
			$get_content      = apply_filters( 'bp_privacy_policy_content', apply_filters( 'the_content', $get_privacy->post_content ), $get_privacy->post_content );
```

**Review by chetansatasiya — CHANGES_REQUESTED** [review]
Need requested changes

**Review by KartikSuthar — APPROVED** [review]

**Pull request merged**